gptel-org.el

```diff
@@ -622,5 +629,44 @@ cleaning up after."
             (prog1 (buffer-substring (point) (point-max))
                    (set-marker start-pt (point-max)))))))))
 
+(defun gptel--org-strip-tool-headers ()
+  "Remove all tool_call block headers and footers."
+  (save-excursion
+    (goto-char (point-min))
+    (while (re-search-forward (rx (or (literal "^#+begin_tool_call")
+                                      (literal "^#+end_tool_call")))
+                              nil t)
+      (delete-region (match-beginning 0) (line-end-position)))))
+
+(defun gptel--org-unescape-tool-results ()
+  "Undo escapes done to keep results from escaping blocks.
+Scans backward for gptel tool text property, reads the arguments, then
+unescapes the remainder."
+  (save-excursion
+    (goto-char (point-max))
+    (let ((prev-pt (point)))
+      (while (> prev-pt (point-min))
+        (goto-char
+         (previous-single-char-property-change (point) 'gptel))
+        (let ((prop (get-text-property (point) 'gptel))
+              (backward-progress (point)))
+          (when (eq (car-safe prop) 'tool)
+            ;; User edits to clean up can potentially insert a tool-call header
+            ;; that is propertized.  Tool call headers should not be
+            ;; propertized.
+            (when (looking-at "[[:space:]]*#\\+begin_tool_call")
```

Comment:
Use `looking-at-p`.  I'm not sure if preserving match data matters here but I don't want to try to find out either.